### DISCUSSION TITLE
Extend Session Time for Large Uploads

### DISCUSSION THREAD

**Discussion by usaboxer** [filebrowser/filebrowser#3151] [General]
Filebrowser seems to timeout around two hours. When uploading a 20gib file over a slow connection it estimates 3-4 hours. It drops connection about halfway through.

A simple input for this in the settings would be a great fix.

I cannot figure out how to make the adjustment manually. If there is an easy way that would be ok too!

I searched for a solution and have only found another question on this issue that was not answered.

**Comment by fastily** [comment]
The [official docs](https://filebrowser.org) are comically out of date (2y at the time of writing). Found #2845 which seems to add a flag `--token-expiration-time` that can be used to set the timeout (e.g. `24h`).  Full disclosure, I know very little golang, but when I tried running the program with this flag it didn't crash immediately

**Reply by collatdam** [reply]
How do you set this flag?

**Reply by fastily** [reply]
it's a command line flag

**Reply by collatdam** [reply]
Understood. What if filebrowser is running in the  container (CasaOS)?

**Reply by derixithy** [reply]
Can this be set in the default docker config? You can run `filebrowser --token-expiration-time 24h` in docker but it's missing it's config.

**Comment by DarthJahus** [comment]
By default, File Browser sessions last around 2 hours. For large downloads, or for convenience, you can extend the session by increasing the JWT token expiration with the `--token-expiration-time` flag.

## Docker / s6
In your `docker-compose.yml`, add:

```yaml
command: s6-setuidgid abc filebrowser -c /config/settings.json --token-expiration-time 48h
```

This preserves s6 init, permissions, and config (check file: [docker/s6/etc/services.d/filebrowser/run](https://github.com/filebrowser/filebrowser/blob/master/docker/s6/etc/services.d/filebrowser/run))

**Comment by 67373net** [comment]
found the official solution, in case anyone needs it:

> [Session Timeout](https://filebrowser.org/troubleshooting.html#session-timeout)
> By default, user sessions expire after 2 hours. If you’re uploading large files over slower connections, you may need to increase this timeout to prevent sessions from expiring mid-upload. You can configure the session timeout using the tokenExpirationTime setting.
> 
> You can either set this option during runtime by using the flag --tokenExpirationTime, the environment variable FB_TOKEN_EXPIRATION_TIME, or in your configuration file. If you want to persist this to the configuration, please use [filebrowser config set](https://filebrowser.org/cli/filebrowser-config-set.html).
> 
> Valid duration formats include "2h", "30m", "24h", or combinations like "2h30m".
